service/history/commandChecker.go

```diff
@@ -485,30 +486,60 @@ func (v *commandAttrValidator) validateContinueAsNewWorkflowExecutionAttributes(
 	}
 	attributes.TaskQueue = taskQueue
 
-	// Reduce runTimeout if it is going to exceed WorkflowExpirationTime
-	// Note that this calculation can produce negative result
-	// handleCommandContinueAsNewWorkflow must handle negative runTimeout value
-	timeoutTime := timestamp.TimeValue(executionInfo.WorkflowExecutionExpirationTime)
-	if !timeoutTime.IsZero() {
-		runTimeout := timestamp.RoundUp(timeoutTime.Sub(time.Now().UTC()))
-		if timestamp.DurationValue(attributes.GetWorkflowRunTimeout()) > 0 {
-			runTimeout = timestamp.MinDuration(runTimeout, timestamp.DurationValue(attributes.GetWorkflowRunTimeout()))
-		} else {
-			runTimeout = timestamp.MinDuration(runTimeout, timestamp.DurationValue(executionInfo.WorkflowRunTimeout))
+	// TODO remove this block after 1.5
+	if !enableInfiniteTimeout() {
```

Comment:
Probably we don't need to have this code path as the only downside is we won't have this adjustment logic during the release.  Which is an ok compromise.